### PR TITLE
miscellaneous

### LINKED ISSUE #36: Timelines are missing some fields
See failing regression tests on master: missing Log, LogFilePath, ModelFile, Seed in the timelines dataframes. How much of this information should be included?

### LINKED ISSUE #37: doc-string consistency - google style

### PR DESCRIPTION
**Pull request by maliheha**
Closes #36 
Close #37 

This PR:
- Adds the cols `ModelFile`, `Model`, `Log`, `LogFilePath`, `LogNumber`, `Seed`, and `Version` to the progress dfs
- Adds one more entry to the progress list of the `NodeLogParser`
- Adds extra simplex iterations of the crossover to the progress list of the `ContinuousParser`. This is what the current implementation does
- The current implementation of the `grblogtools` does not add an extra line of summary to the progress lists of the simplex or barrier root logs. The same strategy is followed here
- Moves the lines capturing the objective value, best bound, and gap (either for MIP or LP) to `NodeLogParser` or `ContinuousParser`
- Adds three more log examples to `assets`
- Adjusts the tests given the changes made
- Applies some cosmetic changes

### CODE REVIEW

File: src/grblogtools/parsers/continuous.py [hunk 1]
```diff
@@ -58,14 +73,15 @@ def parse(self, line: str) -> bool:
...
             matched = self._barrier_parser.parse(line)
-            # If the barrier gets interrupted during the concurrent, switch to
-            # the simplex.
-            if not matched and ContinuousParser.barrier_interruption_pattern.match(
-                line
+            # If the barrier gets interrupted during the concurrent or there are
+            # extra simplex iterations, switch to simplex
+            if not matched and (
+                ContinuousParser.barrier_interruption_pattern.match(line)
+                or SimplexParser.simplex_start_pattern.match(line)
```
Comment (simonbowly) [inline src/grblogtools/parsers/continuous.py]: Can we defer this to SimplexParser.parse()?

File: src/grblogtools/api.py [hunk 2]
```diff
@@ -90,6 +84,23 @@ def summary(self):
                 Log=lambda df: df.apply(strip_model_and_seed, axis=1),
             )
         )
+
```
Comment (simonbowly) [inline src/grblogtools/api.py]: Setting this `_common` attribute would seem to require that summary() is always called before progress()?

File: src/grblogtools/parsers/nodelog.py [hunk 3]
```diff
@@ -93,14 +87,38 @@ def parse(self, line: str) -> bool:
...
+                    "Time": entry["Runtime"],
+                }
+            )
             return True
+
+        match = self.tree_search_termination.match(line)
+        if match:
+            entry = typeconvert_groupdict(match)
+            self._summary.update(entry)
+            self._progress_last_entry.update(
```
Comment (simonbowly) [inline src/grblogtools/parsers/nodelog.py]: I think it's simpler to populate this line in get_progress() rather than tracking another variable attribute.

**Review by simonbowly — COMMENTED** [review]
Looks great, thanks for all the work! I proposed a couple of changes in 5701809

File: src/grblogtools/parsers/continuous.py (hunk 1 above)
Comment (maliheha) [reply src/grblogtools/parsers/continuous.py]: I think you have already done it in this [commit](https://github.com/simonbowly/grblogtools/commit/5701809e2343e5214d9d8b405e31f86c55b25624).

File: src/grblogtools/api.py (hunk 2 above)
Comment (maliheha) [reply src/grblogtools/api.py]: In the current `get_dataframe` function, there is no option to return `progress` without calling `summary`. But you are right. I think you already took care of this in this [commit](https://github.com/simonbowly/grblogtools/commit/783eb28033c877d60b82219cf3e7ed3b1e8da2f6).

File: src/grblogtools/parsers/nodelog.py (hunk 3 above)
Comment (maliheha) [reply src/grblogtools/parsers/nodelog.py]: Yes, I saw your changes. I thought this way is more clear where each pieces is added, but your version is shorter and cleaner with the disadvantage of creating a coy of `self._progress` every time we call the `get_progress()` method on the NodeLogParser.

**Comment by maliheha** [comment]
I just made[ this commi](https://github.com/simonbowly/grblogtools/pull/38/commits/0158d0cc49f60ca5ec3b370b0e373d950f0ffc60)t to cache summary but it works for Python >=3.8. If it is super important to support Python 3.7, we can revert it back. 

Otherwise, thanks for the changes. I think we can merge it after we make the decision about Python 3.7 and whether we want to cache summary or not.

**Comment by simonbowly** [comment]
From Matthias' earlier comments I don't think dropping 3.7 will fly just yet :) The cache is also pretty risky for a couple of reasons:

- caching a mutable return structure like a DataFrame is very problematic (if the user modifies it, all future returns are also modified). This is ok for just `get_dataframe(...)` but not for using `result = parse(...); result.summary()` directly.
    - preferable option would be to cache common_log_data and mark it as only internal
- deleting from `__dict__` is interesting but it seems a bit of an ugly hack ... if caching return values like this I would suggest just `if _cached_thing is None: ...` rather than introducing a secondary method just in order to use the decorator.

Can we strip 0158d0cc for the time being? I left the note more so we are aware of it for future reference, I would not bother with it unless there is a noticeable performance problem. We can leave this on a branch to revisit when 3.7 goes out of support.

**Comment by maliheha** [comment]
> From Matthias' earlier comments I don't think dropping 3.7 will fly just yet :) The cache is also pretty risky for a couple of reasons:
> 
> * caching a mutable return structure like a DataFrame is very problematic (if the user modifies it, all future returns are also modified). This is ok for just `get_dataframe(...)` but not for using `result = parse(...); result.summary()` directly.
>   
>   * preferable option would be to cache common_log_data and mark it as only internal
> * deleting from `__dict__` is interesting but it seems a bit of an ugly hack ... if caching return values like this I would suggest just `if _cached_thing is None: ...` rather than introducing a secondary method just in order to use the decorator.
> 
> Can we strip [0158d0c](https://github.com/simonbowly/grblogtools/commit/0158d0cc49f60ca5ec3b370b0e373d950f0ffc60) for the time being? I left the note more so we are aware of it for future reference, I would not bother with it unless there is a noticeable performance problem. We can leave this on a branch to revisit when 3.7 goes out of support.

I reverted the commit.

**Comment by maliheha** [comment]
You can merge it if you are happy with it. Thanks.

**Pull request merged**